query/src/predicate.rs

```diff
@@ -175,3 +175,32 @@ impl PredicateBuilder {
         self.inner
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_timestamp_range_contains() {
+        let range = TimestampRange::new(100, 200);
+        assert!(!range.contains(99));
+        assert!(range.contains(100));
+        assert!(range.contains(101));
+        assert!(range.contains(199));
+        assert!(!range.contains(200));
+        assert!(!range.contains(201));
+    }
+
+    #[test]
+    fn test_timestamp_range_contains_opt() {
+        let range = TimestampRange::new(100, 200);
+        assert!(!range.contains_opt(Some(99)));
+        assert!(range.contains_opt(Some(100)));
+        assert!(range.contains_opt(Some(101)));
+        assert!(range.contains_opt(Some(199)));
+        assert!(!range.contains_opt(Some(200)));
+        assert!(!range.contains_opt(Some(201)));
+
+        assert!(!range.contains_opt(None));
+    }
+}
```

Comment:
An alternative suggestion:

```suggestion
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_timestamp_range_contains() {
        let range = TimestampRange::new(100, 200);
        assert!([100, 101, 199].iter().all(|&ts| range.contains(ts)));
        assert!([99, 200, 201].iter().all(|&ts| !range.contains(ts)));
    }

    #[test]
    fn test_timestamp_range_contains_opt() {
        let range = TimestampRange::new(100, 200);
        assert!([100, 101, 199]
            .iter()
            .all(|&ts| range.contains_opt(Some(ts))));
        assert!([99, 200, 201]
            .iter()
            .all(|&ts| !range.contains_opt(Some(ts))));

        assert!(!range.contains_opt(None));
    }
}
```

Reply:
While starting to implement this change with a fresh pair of eyes, I conclude that I personally prefer the more verbose form of the test. 

My rational:  if this test ever fails, the more verbose form will tell you exactly which value failed. 

With the more compact formulation, for example, if this fails:
```
        assert!([100, 101, 199].iter().all(|&ts| range.contains(ts)));
```

All you will know is that it was either 100, 101, or 199.

Thus I plan to not implement this suggestion at this time.